parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/state/ShardManager.java

```diff
@@ -132,9 +139,12 @@ ShardKey computeShardKey(ConsumerRecord<?, ?> wc) {
      * @return Work ready in the processing shards, awaiting selection as work to do
      */
     public long getNumberOfWorkQueuedInShardsAwaitingSelection() {
+        // all available container count - (still pending for running retry containers count)
+        // => all_available_count - (retryCnt - all_expired_retry_cnt)
+
         return processingShards.values().stream()
                 .mapToLong(ProcessingShard::getCountOfWorkAwaitingSelection)
-                .sum();
+                .sum() - retryItemCnt.get() + getExpiredItemCntFromRetryQueue();
     }
 
     public boolean workIsWaitingToBeProcessed() {
```

Comment:
refactor to just 
```
public boolean workIsWaitingToBeProcessed() {
    return getNumberOfWorkQueuedInShardsAwaitingSelection() > 0;
}
```
As it needs still to take into account retry queue state for this as well.
in addition remove `workIsWaitingToBeProcessed()` in the ProcessingShard class completely as it cannot be accurate without doing either full scan on all Work Containers in the shard - or needs access to retry queue in the ShardManager. As the method is only used from ShardManager - it is safe to just remove it.